watcher/local/watcher_test.go

```diff
@@ -280,6 +297,66 @@ func Test_Watcher_Working(t *testing.T) {
 
 			rs.AssertExpectations(t)
 		})
+
+		// nolint: dupl
+		t.Run("happy/progressed_event", func(t *testing.T) {
+			parentParams, parentTxs := randomTxsForSingleCh(rng, 3)
+			childParams, childTxs := randomTxsForSingleCh(rng, 3)
+			parentTxs[2].Allocation.Locked = []channel.SubAlloc{{ID: childTxs[0].ID}} // Add sub-channel to allocation.
+
+			adjSubParent, triggerParent := setupAdjudicatorSub(makeProgressedEvent(parentTxs[2])...)
+			adjSubChild, triggerChild := setupAdjudicatorSub(makeProgressedEvent(childTxs[2])...)
+
+			rs := &mocks.RegisterSubscriber{}
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSubParent, nil).Once()
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSubChild, nil).Once()
+
+			w := newWatcher(t, rs)
+			// Parent: Start watching and publish states.
+			parentSignedState := makeSignedStateWDummySigs(parentParams, parentTxs[0].State)
+			statesPubParent, eventsForClientParent := startWatchingForLedgerChannel(t, w, parentSignedState)
+			require.NoError(t, statesPubParent.Publish(parentTxs[1]))
+
+			// Child: Start watching and publish states.
+			childSignedState := makeSignedStateWDummySigs(childParams, childTxs[0].State)
+			statesPubChild, eventsForClientChild := startWatchingForSubChannel(t, w, childSignedState, parentTxs[0].State.ID)
+			require.NoError(t, statesPubChild.Publish(childTxs[1]))
```

Comment:
Both state publishing also not needed here AFAIK.

Reply:
Updated in d20c6b0.